### PR TITLE
[FEATURE] allow geometryless memory layers

### PR DESCRIPTION
**Pull request by SebDieBln**
This PR adds the option to create a memory layer without geometries.
It includes a test for this new functionality.

qgismemorylayergeometryless

### CODE REVIEW

File: tests/src/python/test_provider_memory.py [hunk 1]
```diff
@@ -73,6 +73,10 @@ def testPointCtor(self):
         layer = QgsVectorLayer("Point", "test", "memory")
         assert layer.isValid(), "Failed to create valid point memory layer"
 
+    def testNoGeometryCtor(self):
+        layer = QgsVectorLayer("None", "test", "memory")
+        assert layer.isValid(), "Failed to create valid geometryless memory layer"
```
Comment (nyalldawson) [inline tests/src/python/test_provider_memory.py]: Would be a good idea to check the geometry type on the layer here (and in line 74) too

**Comment by nyalldawson** [comment]
@SebDieBln Looks good! Don't forget to update the dialog tab order when you add widgets like this.

File: tests/src/python/test_provider_memory.py (hunk 1 above)
Comment (SebDieBln) [inline tests/src/python/test_provider_memory.py]: Checks for `geometryType()` and `wkbType()` are in `testLayerGeometry()`, but you are right, _NoGeometry_ was not checked there.

**Comment by SebDieBln** [comment]
I extended the existing tests `testCtors()` and `testLayerGeometry()` to test for every allowed geometry type.

**Comment by SebDieBln** [comment]
@nyalldawson 
> Don't forget to update the dialog tab order when you add widgets like this.

Hm, I already did update it in my original commit. In the UI-file the tab order is correct.
Seems I hit two bugs here:
* When redefining tab-stops in QtCreator the _QDialogButtonBox_ does not get an explicit tab stop, it implicitly becomes the last stop. Since it was explicitly set to be the last tab stop this shouldn't make any difference.
* The `QgsProjectionSelectionWidget` misbehaves regarding to tab stops. This affects the _New Vector Layer_ dialog, too. In some cases the behaviour is even different depending on whether the _QDialogButtonBox_ is an implicit or explicit tab-stop. I am investigating this.

**Pull request merged**